src/cargo/ops/cargo_fetch.rs

```diff
@@ -57,6 +58,34 @@ pub fn fetch<'a>(
             .map(|(id, _deps)| id);
         deps_to_fetch.extend(deps);
     }
+
+    // If -Zbuild-std was passed, download dependencies for the standard library.
+    if let Some(crates) = &config.cli_unstable().build_std {
+        if build_config.build_plan {
+            config
+                .shell()
+                .warn("-Zbuild-std does not currently fully support --build-plan")?;
+        }
+        if build_config.requested_kinds[0].is_host() {
+            // TODO: This should eventually be fixed. Unfortunately it is not
+            // easy to get the host triple in BuildConfig. Consider changing
+            // requested_target to an enum, or some other approach.
+            anyhow::bail!("-Zbuild-std requires --target");
+        }
```

Comment:
IIRC this check also happens on `cargo build`, would you be up for seeing if this requirement could be consolidated somewhere instead of being duplicated here as well?

Reply:
On a hunch, I actually tried removing this check altogether and it seemed to work fine :) for `build` too. I've pushed that change so the whole test suite runs, do you know a reason why the check should still be there?